src/standalone/Vegetation/pfts.jl

```diff
@@ -0,0 +1,491 @@
+"""
+This file contains the PFT struct used to prescribe vegetation parameters to the 
+model when running using PFTs, as well as the default PFT scheme. ClimaLand 
+supports running using default PFTs, user defined PFTs, or fully prescribed 
+vegetation parameters.
+
+The default PFTs used in the model use parameter values drawn from defferent 
+sources in the literature. The sources are listed here with abbreviations that
+are used in the comments in the code below.
+
+    Chen, J. M., G. Mo, J. Pisek, J. Liu, F. Deng, M. Ishizawa, 
+    and D. Chan (2012), Effects of foliage clumping on the estimation of global
+    terrestrial gross primary productivity, Global Biogeochem. Cycles, 26, 
+    GB1019, doi:10.1029/2010GB003996.
+    abb. as Chen et al. 2012
+
+    CLM5.0 Technical Note
+    abb. as CLM5.0
+
+    CLM2 PFT Data
+    https://svn-ccsm-inputdata.cgd.ucar.edu/trunk/inputdata/lnd/clm2/pftdata/
+    abb. as CLM2 PFT Data
+"""
+
+export Pft, default_pfts, params_from_pfts
+
+using DataFrames
+
+# List of parameters that PFTs need to define in order to be valid
+param_list = [
+    # TwoStreamModel parameters
+    :Ω,          # Clumping Index
+    :α_PAR_leaf, # PAR Leaf Reflectance
+    :α_NIR_leaf, # Near-infrared Leaf Reflectance
+    :τ_PAR_leaf, # PAR Leaf Transmittance
+    :τ_NIR_leaf, # Near-infrared Leaf Transmittance
+
+    # Conductance Model
+    :g1, # Medlyn CO2 Sensitivity
+
+    # Photosynthesis model
+    :Vcmax25, # Max rate of carboxylation of Rubisco
+
+    # Plant Hydraulics and general plant parameters
+    :SAI,             # Avg Stem Area Index
+    :f_root_to_shoot, # Root to Shoot Ratio
+    :K_sat_plant,     # Hydraulic Conductivity at Saturation
+    :ψ63,             # Xylem water potential at 63% loss of conductivity
+    :capacity,        # Water Storage Capacity
+    :rooting_depth,    # Plant Rooting Depth
+]
+
+
+"""
+Define a PFT type that can be used to store the parameters for each PFT.
+Each PFT has a name and a list of parameters. The parameters are stored as 
+NamedTuples mapping parameter names to values. The inner constructor checks to
+ensure all required parameters are defined for the PFT.
+"""
+struct Pft
+    name::String
+    parameters::NamedTuple
+
+    # Inner constructor to verify all required parameters are defined
+    function Pft(name, parameters)
+        defined_param_names = keys(parameters)
+        for param in param_list
```

Comment:
Im not sure if it matters, but is param_list globally accessible? should it be defined within the constructor?

Reply:
My thinking was that we should export the param_list from the pft module so that if a user wants to know which parameters are defined by PFT structs then they could potentially view that list but if you don't think that useful I'm happy to move it into the constructor.

Reply:
I was maybe confused and thinking of the case of a script that has something like:
```
a = [1,2,3]
function this_gets_called_every_timestep(x)
    x .= a
end
```
which would have a slowdown because `a` is defined only in global scope. But I think that isnt the same case when something is defined within a module, you have with` param_list`? Also, Pft() is only called once, not every step, but still would be good to avoid if it *is* an issue.